server/mscalendar/availability.go

```diff
@@ -138,24 +140,53 @@ func (m *mscalendar) GetAvailabilities(remoteUserID string, scheduleIDs []string
 
 func (m *mscalendar) setStatusFromAvailability(mattermostUserID, currentStatus string, av remote.AvailabilityView) string {
 	currentAvailability := av[0]
+	u := NewUser(mattermostUserID)
+
+	if !m.ShouldChangeStatus(u) {
+		return "Status not changed by user configuration."
+	}
+
+	if !m.HasAvailabilityChanged(u, currentAvailability) {
+		return "Status not changed because there is no update since last status change."
+	}
+
+	u.LastStatusUpdateAvailability = currentAvailability
```

Comment:
I think we need to store more information here to take into account for the user manually changing their status. Here's a proposal

When the user manually changes their status, we take all events that overlap with `now`, and store them as hashes `(eventid-starttime)`. Those events are now "immune" to an availability change. If an event's start gets moved, it becomes a "new" event as far as the availability changer is concerned.

I think this solves the one-off issue when the user wants to cancel out their most recent status change. If they want it to remain untouched by the automation, then they can disable it in their settings.

Reply:
One of the main problems is that there is no eventID that I can use to handle this, I can only work with times and availability.

I can do the following:
- Check if the availability has changed. If changed:
  - Check that mattermost status and handle it
- If the availability is not changed, check the latest start time registered, and compare it with the elements returned.
- If there is a new event (an event that starts later than the last registered) check with the current mattermost status.
- If no event starts later, we assume it is the same event and ignore it.

I will implement this and keep thinking about it. Probably in code make more sense.

Reply:
Hmm well if we don't have an event id we're kind of at a disadvantage in general. Maybe we shouldn't even be using `GetSchedule` since we aren't getting the performance we were getting before with it, which was looking at several users' schedules in one request, the main reason why we had the dedicated bot account.

`GetSchedule`'s main purpose is, find some time where one or more of these people are free given some time constraints. That's not really what we're doing here. We just want to know if they're free "right now". We might want to use `GetCalendarView` with a small time window for this instead.

@larkox Let me know what you think about the shortcomings of using `GetSchedule`, versus the alternative `GetCalendarView`. Which reminds me that we have a `GetSchedule` response json example in the `server/testdata` folder, but not one for `GetCalendarView`.